frontend/ui/src/pages/Inbox/Messenger/ConversationMetadata/index.tsx

```diff
@@ -81,13 +112,32 @@ const ConversationMetadata = (props: ConnectedProps<typeof connector>) => {
     removeTagFromConversation(conversation.id, tag.id);
   };
 
+  const removeNote = (note: NoteModel) => {
+    removeNoteFromConversation(conversation.id, note.id);
+  };
+
+  const updateNote = (note: NoteModel) => {
+    setNoteText(note.text);
+    setShowNotesDialog(true);
+    setNoteId(note.id);
+  };
+
   const tagSorter = (a: TagModel, b: TagModel) => a.name.localeCompare(b.name);
 
-  const conversationTags = () =>
-    Object.keys(conversation.metadata.tags || {})
+  const conversationTags = () => {
+    return Object.keys(conversation.metadata.tags || {})
       .map(tagId => tags[tagId])
       .filter(tag => tag !== undefined)
       .sort(tagSorter);
+  };
+
+  const conversationNotes = () => {
```

Comment:
Refactoring idea: Let's move both `conversationTags` and `conversationNotes` to `lib/typescript/model/Conversation.ts` so that we keep data logic close to the types.